woocommerce/woocommerce-ios
# Bump deployment target to 15.0

Closes: #7850


### Description


Updates the minimum deployment version to 15.0 on all the project's targets. Doing this shows 40+ new deprecation warnings. I've addressed a couple of them but others will need some extra work:

- [x] 'animation' was deprecated in iOS 15.0: Use withAnimation or animation(_:value:) instead.
- [x] 'kUTTypeImage'/'kUTTypeGIF' was deprecated in iOS 15.0
- [ ] 'contentEdgeInsets'/'imageEdgeInsets'/'titleEdgeInsets' was deprecated in iOS 15.0: This property is ignored when using UIButtonConfiguration. There's a [whole new way of styling buttons on iOS 15](https://sarunw.com/posts/new-way-to-style-uibutton-in-ios15/), which would require more than a quick fix
- [ ] 'windows' was deprecated in iOS 15.0: Use UIWindowScene.windows on a relevant window scene instead. This is mostly used on a reimplementation of `currentKeyWindow`, which was deprecated in iOS 13. We'll need to eventually invest some time in adapting the existing hacks to window scenes, or removing them altogether.

Also noted ☝🏽 in p91TBi-9WX-p2

### Testing instructions


CI tests should pass.

For the animation change:
1. Go to Settings > Help & Support > View Application Log
2. The button on the bottom-right corner should scroll to the bottom of the log with an animation

For the UTTType changes:
1. Save a GIF to your camera roll
2. Go to products
3. Add a new product
4. Add an image, select "Choose from device"
5. Select the "Recents" album
6. The GIF should have a "GIF" badge.

---
